components/omega/src/ocn/Eos.h

```diff
@@ -0,0 +1,257 @@
+#ifndef OMEGA_EOS_H
+#define OMEGA_EOS_H
+//===-- ocn/Eos.h - Equation of State --------------------*- C++ -*-===//
+//
+/// \file
+/// \brief Contains functors for calculating specific volume
+///
+/// This header defines functors to be called by the time-stepping scheme
+/// to calculate the specific volume based on the choice of EOS
+//
+//===----------------------------------------------------------------------===//
+
+#include "AuxiliaryState.h"
+#include "Config.h"
+#include "EosConstants.h"
+#include "HorzMesh.h"
+#include "MachEnv.h"
+#include "OmegaKokkos.h"
+#include "TimeMgr.h"
+#include <string>
+
+namespace OMEGA {
+
+enum class EosType {
+   Linear,       /// Linear equation of state
+   TEOS10Poly75t /// Roquet et al. 2015 75 term expansion
+};
+
+//
+/// TEOS10 75-term polynomial
+class TEOS10Poly75t {
```

Comment:
Do we actually need individual classes for each option?  I think TEOS is going to be the default nearly all the time with Linear as a trivial special case option. So may as well keep a single Eos class.

Reply:
@philipwjones I have consolidated to a single class in this commit: https://github.com/E3SM-Project/Omega/pull/222/commits/96a7ac94c5862283954e0a5ff3c9deaaf3dbae96, and I would like your feedback.

Tests pass on pm-cpu and pm-gpu, but I would like to know if this fits what you were suggesting here or if there are any new comments/edits you have that apply to this new structuring. I know that I still need to move calcDisplacedSpecVol to be public and make a few changes with regards to that, but comments on the overall implementation would be helpful as I get use to developing in Omega.

Reply:
@katsmith133 Yes, that looks perfect.